### ISSUE TITLE
core: should use setpgrp() and killpg() on shutdown.

### ISSUE THREAD

**Issue by dw** [dw/mitogen#264]
During broker shutdown, we should kill any children belonging to our process group. Presently on adding this code, on Linux sshd immediately receives a SIGCHLD when we call setpgrp().

**Comment by wichtounet** [comment]
Any news on that issue? It would be great to be able to kill the children on shutdown.

**Comment by dw** [comment]
Hey there,

You should not be seeing stray processes after a run completes. Is this happening for you? If so, please open a new bug and answer the questions from the template: )

Thanks, David

**Comment by wichtounet** [comment]
Yes, we still see stray remote processes after a shutdown() if the commands are still running. We are going to investigate further and open a new issue if necessary. 

Thanks for the quick answer.

**Comment by dw** [comment]
`MITOGEN_ROUTER_DEBUG=1` will often catch exceptions occurring late during shutdown and write them to `/tmp/mitogen.*.log` on the target machine, but I just noticed it is possibly broken on current master. It should be fine in the previous release
